### PR TITLE
DIRKRB-745

### PR DESCRIPTION
**Pull request by liujiayi771**
Add command tool ktadd(xst) in remote kadmin tool.

### CODE REVIEW

File: kerby-kerb/kerb-admin-server/src/main/java/org/apache/kerby/kerberos/kerb/admin/server/kadmin/AdminServerHandler.java [hunk 1]
```diff
@@ -22,23 +22,19 @@
...
 import org.apache.kerby.kerberos.kerb.admin.kadmin.local.LocalKadmin;
 import org.apache.kerby.kerberos.kerb.admin.kadmin.local.LocalKadminImpl;
-import org.apache.kerby.kerberos.kerb.admin.message.AddPrincipalRep;
-import org.apache.kerby.kerberos.kerb.admin.message.AdminMessage;
-import org.apache.kerby.kerberos.kerb.admin.message.AdminMessageCode;
-import org.apache.kerby.kerberos.kerb.admin.message.AdminMessageType;
-import org.apache.kerby.kerberos.kerb.admin.message.DeletePrincipalRep;
-import org.apache.kerby.kerberos.kerb.admin.message.GetprincsRep;
-import org.apache.kerby.kerberos.kerb.admin.message.KadminCode;
-import org.apache.kerby.kerberos.kerb.admin.message.RenamePrincipalRep;
```
Comment (plusplusjiajia) [inline kerby-kerb/kerb-admin-server/src/main/java/org/apache/kerby/kerberos/kerb/admin/server/kadmin/AdminServerHandler.java:32]: It's better not to use * for import.

File: kerby-kerb/kerb-admin-server/src/main/java/org/apache/kerby/kerberos/kerb/admin/server/kadmin/AdminServerHandler.java [hunk 2]
```diff
@@ -199,6 +199,42 @@ private ByteBuffer handleGetprincsReq(LocalKadmin localKadmin, XdrFieldInfo[] fi
...
     }
+    
+    private ByteBuffer handleExportKeytabReq(LocalKadmin localKadmin, XdrFieldInfo[] fieldInfos) throws IOException {
+        String principals = ((String) fieldInfos[2].getValue());
+        
+        if (principals != null) {
+            List<String> princList = stringToList(principals);
+            if (princList.size() != 0) {
+                LOG.info("Exporting keytab file for " + principals + "...");
+                File path = new File("/tmp/" + System.currentTimeMillis());
```
Comment (dwalluck) [inline kerby-kerb/kerb-admin-server/src/main/java/org/apache/kerby/kerberos/kerb/admin/server/kadmin/AdminServerHandler.java]: I just happened to notice this. It is not the correct temporary directory on Windows and it is also not secure. Consider using `File.createTempFile()` instead.

Comment (liujiayi771) [reply kerby-kerb/kerb-admin-server/src/main/java/org/apache/kerby/kerberos/kerb/admin/server/kadmin/AdminServerHandler.java]: I have fixed it by using Files.createTempDirectory("ktadd").

File: kerby-kerb/kerb-admin-server/src/main/java/org/apache/kerby/kerberos/kerb/admin/server/kadmin/AdminServerHandler.java (hunk 1 above)
Comment (liujiayi771) [reply kerby-kerb/kerb-admin-server/src/main/java/org/apache/kerby/kerberos/kerb/admin/server/kadmin/AdminServerHandler.java:32]: I have fixed it, and I also added the UT of the remote kadmin server.

File: kerby-kerb/kerb-admin-server/src/test/java/org/apache/kerby/kerberos/kerb/admin/server/RemoteKadminTest.java [hunk 3]
```diff
@@ -0,0 +1,236 @@
+package org.apache.kerby.kerberos.kerb.admin.server;
```
Comment (plusplusjiajia) [inline kerby-kerb/kerb-admin-server/src/test/java/org/apache/kerby/kerberos/kerb/admin/server/RemoteKadminTest.java:20]: The apache license should be added.

**Review by plusplusjiajia — APPROVED** [review]

**Comment by plusplusjiajia** [comment]
LGTM, I've merged this PR.

**Comment by coheigea** [comment]
@plusplusjiajia Why not just merge the PR via the "Merge pull request" button on this page?

**Comment by plusplusjiajia** [comment]
> @plusplusjiajia Why not just merge the PR via the "Merge pull request" button on this page?

@coheigea I can't see the "Merge pull request" button and "close" button, I think I don't have these permissions.

**Comment by coheigea** [comment]
Ah I see. You need to follow the instructions here about linking your ASF + Github accounts: https://gitbox.apache.org/

**Comment by plusplusjiajia** [comment]
> Ah I see. You need to follow the instructions here about linking your ASF + Github accounts: https://gitbox.apache.org/

Thanks for your guide. :)

**Pull request closed without merging**